nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,730 @@
+        scalar values to store for that particular information $i$.
+    """
+    def __init__(self, streamlines=None,
+                 data_per_streamline=None,
+                 data_per_point=None,
+                 affine_to_rasmm=None):
+        """
+        Parameters
+        ----------
+        streamlines : iterable of ndarrays or :class:`ArraySequence`, optional
+            Sequence of $T$ streamlines. Each streamline is an ndarray of
+            shape ($N_t$, 3) where $N_t$ is the number of points of
+            streamline $t$.
+        data_per_streamline : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            streamline, and its associated value is an iterable of ndarrays of
+            shape ($P_i$,) where $P_i$ is the number scalar values to store
+            for that particular information $i$.
+        data_per_point : dict of iterable of ndarrays, optional
+            Dictionary where the items are (str, iterable).
+            Each key represents an information $i$ to be kept along side every
+            point of every streamline, and its associated value is an iterable
+            of ndarrays of shape ($N_t$, $M_i$) where $N_t$ is the number of
+            points for a particular streamline $t$ and $M_i$ is the number
+            scalar values to store for that particular information $i$.
+        affine_to_rasmm : ndarray of shape (4, 4) or None, optional
+            Transformation matrix that brings the streamlines contained in
+            this tractogram to *RAS+* and *mm* space where coordinate (0,0,0)
+            refers to the center of the voxel. By default, the streamlines
+            are in an unknown space, i.e. affine_to_rasmm is None.
+        """
+        self._set_streamlines(streamlines)
+        self.data_per_streamline = data_per_streamline
+        self.data_per_point = data_per_point
+        self.affine_to_rasmm = affine_to_rasmm
+
+    @property
+    def streamlines(self):
+        return self._streamlines
+
+    def _set_streamlines(self, value):
+        self._streamlines = ArraySequence(value)
+
+    @property
+    def data_per_streamline(self):
+        return self._data_per_streamline
+
+    @data_per_streamline.setter
+    def data_per_streamline(self, value):
+        self._data_per_streamline = PerArrayDict(len(self.streamlines), value)
+
+    @property
+    def data_per_point(self):
+        return self._data_per_point
+
+    @data_per_point.setter
+    def data_per_point(self, value):
+        self._data_per_point = PerArraySequenceDict(
+            self.streamlines.nb_elements, value)
+
+    @property
+    def affine_to_rasmm(self):
+        """ Affine bringing streamlines in this tractogram to RAS+mm. """
+        return copy.deepcopy(self._affine_to_rasmm)
+
+    @affine_to_rasmm.setter
+    def affine_to_rasmm(self, value):
```

Comment:
Should there be a check on the shape of the value, to make sure this is a valid affine? Or we leave that to code further along that will try to use it and fail if it's not correct?